### ISSUE TITLE
Move Tasks and Targets from Microsoft.Packaging.Tools into SDK

### ISSUE THREAD

**Issue by dsplaisted** [dotnet/sdk#974]
The NETStandard.Library repo creates a Microsoft.Packaging.Tools package with tasks and targets to resolve conflicts between different versions of the same assembly, and to support trimming dependencies that are not used.  These should be moved into the SDK repo, as they integrate with what it does, and this is in some cases this is fragile because they are outside the SDK (for example they have to rewrite the deps file after it has already been generated by the SDK).

Some of these tasks and targets also need to be active in non-SDK projects that reference the NETStandard.Library NuGet package.  The best solution is probably to move all the code into the SDK repo, include it in the SDK, and also publish a Microsoft.Packaging.Tools NuGet package out of the SDK that includes these tasks and targets for non-SDK projects.

Tasks:

- [x] Copy tasks and targets into SDK, update them to build as part of SDK tasks
- [x] Add switch to disable logic from existing Microsoft.Packaging.Tools (to turn it off when we're using the SDKs version)
- [x] Hook up targets in SDK, and disable logic from Microsoft.Packaging.Tools in SDK (PR ETA 3/24)
- [ ] Add tests for conflict resolution (PR ETA 3/24)
  - [x] Conflict resolution for Build
  - [x] Conflict resolution for Publish of portable app
  - [x] Conflict resolution for Publish of self-contained app
  - [x] Conflict resolution for .NET Framework projects using SDK
  - [ ] Conflict resolution between a simple and a concrete reference (this is not handled by the conflict resolution target currently)
  - [x] Conflicts with items from PlatformManifest (covered with unit tests)
  - [x] Unit tests on conflicts resolver
  - [ ] Consider validating deps file in tests
    - Verify that there is no duplication
    - For self-contained apps, verify that everything in the deps is published
  - [ ] Conflict resolution for .NET Framework projects not using SDK
- [x] Don't generate message about conflicts that can't be resolved for multiple references to the same framework assembly
- [x] Move hardcoded strings from conflict resolution to resources to support localization
- [ ] Rationalize where downlevel HandlePackageFileConflicts comes from
- [ ] Refactoring / cleanup
  - [ ] Refactor to avoid calling `HandlePackageFileConflicts` twice, once for build and once for publish
  - [ ] Refactor to avoid calling `ResolvePublishAssemblies` twice
  - [ ] Consider using MSBuild items for deps file generation instead of re-reading the assets file.
  - [ ] Use item metadata instead of walking directory tree in `NuGetUtilities.GetPackageParts`

**Comment by dsplaisted** [comment]
Copying the contents of an email from @ericstj about this for easy reference:

To do the work let’s try to stage it so that we have a switch that we can turn off the current stuff when it’s done in the SDK.  Add that switch in standard-repo, do the work in the SDK, once the SDK is done set the switch to disable standard-repo implementation then we go and delete the stuff in standard repo.  We’ll need that switch anyhow to understand when not to run the stuff that remains for downlevel.

At least the HandlePackageFileConflicts task needs to ship down-level.  It’s ok to copy that code for now, do you want to have a shared assembly moving forward?  

I’ve only heard one report of a race condition with no reliable repro.  I did have a hack with how I rewrote the deps.  I did this because SDK was writing directly to the final output location and I knew the standard-repo impl was temporary.  Given that file was just written to read it and write it again will always run into file-in-use issues with things like virus-scanners and search indexers.  Ideally in the SDK this wouldn’t be a rewriting the deps, instead the task that writes the deps initially should take in a list of items to exclude and we can just fill in that list.

A couple other things to look at fixing:
1.	Avoid the two different calls to `HandlePackageFileConflicts`.  I do this for build and for publish.  Can we do it once for both?
2.	Avoid the call to `ResolvePublishAssemblies`.  This is already done elsewhere in SDK targets and we shouldn’t do it twice, but instead make a single call work for both cases.
3.	(somewhat unrelated) Refactor the deps file generation to use items that already flow through MSBuild rather than re-reading the assets file.  This could be a more natural solution for targets to contribute to / modify deps file contents.  It could also be unnecessary complexity so don’t take my word for it.

It’s fine to leave the trimming out.  I think that’s expected.  Let’s just make sure that whatever is responsible for the deps file writing can get the input from the trimming task as it does today.
